src/server/index.js

```diff
@@ -69,7 +69,9 @@ async function NextAuthHandler (req, res, userOptions) {
 
     const { csrfToken, csrfTokenVerified } = csrfTokenHandler(req, res, cookies, secret)
 
-    const providers = parseProviders({ providers: userOptions.providers, baseUrl, basePath })
+    const locale = req.query.locale || undefined
```

Comment:
Think this one needs to be moved to https://github.com/nextauthjs/next-auth/blob/2d9711eeefd2c5826192a0a5e9e30f39264bb1bf/src/server/index.js#L52-L57

As we use a catch-all route.

How would the URLs look like by the way? `/api/auth/en/signin` or something? I have to think more about this.

Reply:
This is what api routes look like with locale data: `http://localhost:3000/api/auth/signin?callbackUrl=url&locale=en`

Locale is tracked with a query param for all /api/auth/ routes. I think that is a better option than implementing it as part of the path.